bundles/org.openhab.binding.magentatv/src/main/java/org/openhab/binding/magentatv/internal/network/MagentaTVPoweroffListener.java

```diff
@@ -0,0 +1,134 @@
+/**
+ * Copyright (c) 2010-2020 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.magentatv.internal.network;
+
+import static org.openhab.binding.magentatv.internal.MagentaTVBindingConstants.*;
+import static org.openhab.binding.magentatv.internal.MagentaTVUtil.*;
+
+import java.io.IOException;
+import java.net.DatagramPacket;
+import java.net.InetAddress;
+import java.net.MulticastSocket;
+import java.net.NetworkInterface;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.binding.magentatv.internal.MagentaTVHandlerFactory;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link MagentaTVPoweroffListener} implements a UPnP listener to detect
+ * power-off of the receiver
+ *
+ * @author Markus Michels - Initial contribution
+ */
+@NonNullByDefault
+public class MagentaTVPoweroffListener extends Thread {
+    private final Logger logger = LoggerFactory.getLogger(MagentaTVPoweroffListener.class);
+
+    private final MagentaTVHandlerFactory handlerFactory;
+
+    public static final String UPNP_MULTICAST_ADDRESS = "239.255.255.250";
+    public static final int UPNP_PORT = 1900;
+    public static final String UPNP_BYEBYE_MESSAGE = "ssdp:byebye";
+
+    protected final MulticastSocket socket;
+    protected @Nullable NetworkInterface networkInterface;
+    protected byte[] buf = new byte[256];
+
+    public MagentaTVPoweroffListener(MagentaTVHandlerFactory handlerFactory,
+            @Nullable NetworkInterface networkInterface) throws IOException {
+        this.handlerFactory = handlerFactory;
+        this.networkInterface = networkInterface;
+        socket = new MulticastSocket(UPNP_PORT);
+    }
```

Comment:
For this thread can you set the name and deamon:
```
setName("OH-Binding-magentatv");
setDeamon(true);
```

Reply:
changed